src/config/auth.rs

```diff
@@ -0,0 +1,19 @@
+use serde::{Deserialize, Serialize};
+
+use crate::auth::jwt::JwtProviderOptions;
+
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq)]
+pub struct Auth {
+  #[serde(rename = "JWT")]
+  pub(crate) jwt: Option<JwtProviderOptions>,
+}
+
+impl Auth {
+  pub fn merge_right(self, other: Auth) -> Self {
+    Self { jwt: self.jwt.or(other.jwt) }
+  }
+
+  pub fn is_some(&self) -> bool {
+    self.jwt.is_some()
+  }
+}
```

Comment:
- Auth into `config.rs`
- Auth should be an enum

**When we allow only one auth provider**
```rust
enum Auth {
  JWT(JwtProviderOptions),
  Basic { token: String }
}
```
```graphql

schema 
  @server(auth: {basic: {token: "{{env.BASIC_AUTH_TOKEN}}"}}) {
  query: Query
}
```

**When we have multiple providers**

```rust
struct Provider {id: String, provider: Auth}
enum Auth {
  JWT(JwtProviderOptions),
  Basic { token: String }
}
```

```graphql
schema 
  @server(auth: [
   {id: "basic-auth-1", provider: {basic: {token: "{{env.BASIC_AUTH_TOKEN}}"}}},
   {id: "basic-auth-2", provider: {basic: {token: "{{env.BASIC_AUTH_TOKEN}}"}}},
  ]) {
  query: Query
}
```